### PR TITLE
Hosted plugin - Add support of killing sub treads in Linux run by shell script

### PR DESCRIPTION
**Pull request by idantr**
Signed-off-by: Idan Treivish <idan.treivish@sap.com>

#### What it does
Kills Linux sub treads started by shell script [link](https://nodejs.org/docs/latest-v10.x/api/child_process.html#child_process_subprocess_kill_signal)

#### How to test
1. Docker with Linux
2. Create yarn sh script that internally calls yarn
3. Insert your yarn folder first in PATH
    PATH should be updated before you start theia
    then you can reproduce the issue in local docker
    and the test is before and after

#### Review checklist

- [ ] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-dev/src/node/hosted-plugins-manager.ts [hunk 1]
```diff
@@ -78,19 +79,28 @@ export class HostedPluginsManagerImpl implements HostedPluginsManager {
...
-        const pluginPath = FileUri.fsPath(uri);
-
-        const watchProcess = this.watchCompilationRegistry.get(pluginPath);
-        if (!watchProcess) {
-            throw new Error('Watcher is not running in ' + pluginPath);
-        }
-
-        watchProcess.kill();
-        return Promise.resolve();
+  private killProcessTree(parentPid: number): void {
```
Comment (akosyakov) [inline packages/plugin-dev/src/node/hosted-plugins-manager.ts]: something fishy with formatting, could you make sure to use Theia or VS Code to format this file

Comment (idantr) [reply packages/plugin-dev/src/node/hosted-plugins-manager.ts]: i will reformat it. load a new commit soon

Comment (idantr) [reply packages/plugin-dev/src/node/hosted-plugins-manager.ts]: done

**Comment by amiramw** [comment]
Tested this PR compared to master in gitpod. It seems that step 2 is not needed as yarn is implemented as sh script already. Without this fix hosted theia leaks those watch processes:
image

With this change everything under yarn watch is killed properly.

**Pull request merged**